src/redisearch_rs/rqe_iterators/src/profile.rs

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) 2006-Present, Redis Ltd.
+ * All rights reserved.
+ *
+ * Licensed under your choice of the Redis Source Available License 2.0
+ * (RSALv2); or (b) the Server Side Public License v1 (SSPLv1); or (c) the
+ * GNU Affero General Public License v3 (AGPLv3).
+*/
+
+//! Profile iterator for collecting performance metrics.
+//!
+//! This module provides a wrapper iterator that collects profiling metrics
+//! (read/skip counts and wall-clock time) from a child iterator without
+//! modifying its behavior.
+
+use std::time::Instant;
+
+use crate::{RQEIterator, RQEIteratorError, RQEValidateStatus, SkipToOutcome};
+use ffi::t_docId;
+use inverted_index::RSIndexResult;
+
+/// Profile counters collected during query execution.
+#[derive(Debug, Default, Clone, Copy)]
```

Comment:
```suggestion
#[derive(Debug, Default, Clone)]
```

We usually don't implement `Copy` on non-trivial types as they may be costly to copy.